api/v1/recipe/views.py

```diff
@@ -92,6 +104,10 @@ class RatingViewSet(viewsets.ReadOnlyModelViewSet):
     def get_queryset(self):
         query = Recipe.objects
 
+        # If user is anonymous, restrict recipes to public.
+        if not self.request.user.is_authenticated:
+            query = query.filter(public=True)
```

Comment:
This should be added to the `filter_set` below. 

Something like this should work:
``` python
filter_set = {}
if not self.request.user.is_authenticated:
    filter_set['public'] = True
```

Chaining together filters can be weird in django.